sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/DocumentResult.java

```diff
@@ -64,4 +67,16 @@ public TextAnalyticsError getError() {
     public boolean isError() {
         return isError;
     }
+
+    /**
+     * Throw a {@link TextAnalyticsException} if result has isError true and when a non-error property was accessed.
+     */
+    void throwExceptionIfError() {
```

Comment:
Since this isn't package private it won't be inherited by the subclasses, will this require any additional parameters to be passed?

Reply:
Given that it is an exception to be raised for accessing success attributes on output models, this method would always be accessed in the model classes which exist in the same packages.